spec/rspec/support/object_inspector_spec.rb

```diff
@@ -67,6 +67,30 @@ def with_date_loaded
         end
       end
 
+      context 'with Hash Objects' do
+        let(:hash) { { 'key' => BigDecimal('3.3') } }
+        let(:formatted_hash) { ObjectInspector.inspect(hash) }
+
+        it 'recursively uses itself to inspect objects within it' do
+          in_sub_process_if_possible do
+            require 'bigdecimal'
+            expect(formatted_hash).to include('=> 3.3 (#<BigDecimal')
+          end
+        end
+      end
+
+      context 'with Array Objects' do
+        let(:array) { [[BigDecimal('3.3')]] }
+        let(:formatted_array) { ObjectInspector.inspect(array) }
+
+        it 'recursively uses itself to inspect objects within it' do
+          in_sub_process_if_possible do
+            require 'bigdecimal'
+            expect(formatted_array).to include('[[3.3 (#<BigDecimal')
+          end
+        end
+      end
```

Comment:
I've fixed it for hashes, but as mentioned in my comments, I think the formatting of Arrays should be left for another ticket pending a discussion about the direction rspec-support wants to take in formatting them via the differ.